Kotest Gradle plugin is missing in snapshot publications

The `io.kotest.multiplatform` Gradle plugin (`io.kotest:kotest-framework-multiplatform-plugin-gradle`) is not published to the [snapshot repo](https://s01.oss.sonatype.org/content/repositories/snapshots/io/kotest/). This complicates configuring Kotest dependencies (separate version for plugin and libraries required) and blocks using snapshots which require an updated plugin.

Hmmm Sonatype doesn't like something about the plugin marker...

https://github.com/kotest/kotest/actions/runs/10017676112/job/27694238728#step:7:2337

```
> Task :kotest-framework:kotest-framework-multiplatform-plugin-gradle:publishKotestMultiplatformCompilerGradlePluginPluginMarkerMavenPublicationToDeployRepository FAILED


FAILURE: Build failed with an exception.
Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.


* What went wrong:
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
Execution failed for task ':kotest-framework:kotest-framework-multiplatform-plugin-gradle:publishKotestMultiplatformCompilerGradlePluginPluginMarkerMavenPublicationToDeployRepository'.

> Failed to publish publication 'KotestMultiplatformCompilerGradlePluginPluginMarkerMaven' to repository 'deploy'
   > Could not PUT 'https://s01.oss.sonatype.org/content/repositories/snapshots/io/kotest/multiplatform/io.kotest.multiplatform.gradle.plugin/5.10.0.1533/io.kotest.multiplatform.gradle.plugin-5.10.0.1533.pom'. Received status code 400 from server: Bad Request

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Get more help at https://help.gradle.org./

BUILD FAILED in 16m 24s
For more on this, please refer to https://docs.gradle.org/8.7/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
1770 actionable tasks: 1505 executed, 265 from cache

Publishing build scan...
https://gradle.com/s/mv2xfqfi6bcia
```

[build scan](https://scans.gradle.com/s/mv2xfqfi6bcia)


Possibly it's missing the Javadoc JAR.

--- a/buildSrc/src/main/kotlin/kotest-publishing-conventions.gradle.kts
+++ b/buildSrc/src/main/kotlin/kotest-publishing-conventions.gradle.kts
@@ -24,9 +24,8 @@ signing {
    if (signingKey != null && signingPassword != null) {
       useInMemoryPgpKeys(signingKey, signingPassword)
    }
-   if (Ci.isRelease) {
-      sign(publishing.publications)
-   }
+   sign(publishing.publications)
+   setRequired { Ci.isRelease } // only require signing when releasing
 }
 
 publishing {

--- a/buildSrc/src/main/kotlin/kotest-watchos-device-conventions.gradle.kts
+++ b/buildSrc/src/main/kotlin/kotest-watchos-device-conventions.gradle.kts
@@ -1,7 +1,6 @@
 import org.jetbrains.kotlin.gradle.ExperimentalKotlinGradlePluginApi
 import org.jetbrains.kotlin.gradle.plugin.KotlinHierarchyTemplate
 
-
 plugins {
    id("kotlin-conventions")
 }

--- a/kotest-framework/kotest-framework-multiplatform-plugin-gradle/build.gradle.kts
+++ b/kotest-framework/kotest-framework-multiplatform-plugin-gradle/build.gradle.kts
@@ -72,7 +72,9 @@ tasks.withType<Test>().configureEach {
    }
 }
 
+@Suppress("UnstableApiUsage")
 gradlePlugin {
+   isAutomatedPublishing = true
    website.set("https://kotest.io")
    vcsUrl.set("https://github.com/kotest")
    plugins {
@@ -80,8 +82,8 @@ gradlePlugin {
          id = "io.kotest.multiplatform"
          implementationClass = "io.kotest.framework.multiplatform.gradle.KotestMultiplatformCompilerGradlePlugin"
          displayName = "Kotest Multiplatform Compiler Plugin"
-         description = "Adds support for Javascript and Native tests in Kotest"
-         tags.set(listOf("kotest", "kotlin", "testing", "integrationtesting", "javascript"))
+         description = "Adds support for JavaScript and Native tests in Kotest"
+         tags.addAll("kotest", "kotlin", "testing", "integration testing", "javascript", "native")
       }
    }
 }

